activerecord/lib/active_record/connection_adapters/schema_cache.rb

```diff
@@ -101,20 +111,20 @@ def clear_data_source_cache!(name)
         @columns_hash.delete name
         @primary_keys.delete name
         @data_sources.delete name
+        @indexes.delete name
       end
 
       def marshal_dump
         # if we get current version during initialization, it happens stack over flow.
         @version = connection.migration_context.current_version
-        [@version, @columns, @columns_hash, @primary_keys, @data_sources]
+        [@version, @columns, @columns_hash, @primary_keys, @data_sources, @indexes]
       end
 
       def marshal_load(array)
```

Comment:
What still uses `marshal_dump` and `marshal_load`? Can they fully be supported with `encode_with` and `init_with` and be removed?

Reply:
I think nothing. We can remove that probably.